go/tendermint/pubsub.go

```diff
@@ -0,0 +1,68 @@
+package tendermint
+
+import (
+	"github.com/eapache/channels"
+
+	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+var (
+	_ tmtypes.Subscription = (*tendermintPubsubBuffer)(nil)
+)
+
+type tendermintPubsubBuffer struct {
```

Comment:
Even though these are internal things, can you add some documentation comments around the struct that explains why this is needed and what it does (wrapping to avoid deadlocks in tendermint pubsub).